extension/src/frontend/recording.ts

```diff
@@ -0,0 +1,254 @@
+import { BrowserEvent } from '@/schemas/recording'
+
+import { BrowserExtensionClient } from '../messaging'
+import { generateSelector } from '../selectors'
+import {
+  findAssociatedElement,
+  findInteractiveElement,
+  isNativeButton,
+  isNativeCheckbox,
+  isNativeRadio,
+} from '../utils/dom'
+
+import { WindowEventManager } from './manager'
+import { getTabId } from './utils'
+
+export function startRecording(client: BrowserExtensionClient) {
```

Comment:
This is essentially the same code that was in `extension/src/frontend/index.ts`, except:

- It's been wrapped in a function
- The `tab` property of recorded events use `getTabId` instead of `''`.